upload/upload.go

```diff
@@ -144,6 +147,45 @@ func CheckTokenExpiration(accessToken string) (bool, error) {
 	return tomorrow.After(expiration), nil
 }
 
+// progress bar stuff
```

Comment:
I'd like some more documentation about what this stuff actually does! 
Also - in case we wanna add a progress bar to download as well, we might wanna have the CustomReader in helpers (or it's own file).

Reply:
I'll move it in helpers for a start and add a better description.

Reply:
It's moved, Some more comments added too.